controllers/ibmpowervscluster_controller_test.go

```diff
@@ -294,24 +309,318 @@ func TestIBMPowerVSClusterReconciler_Reconcile(t *testing.T) {
+							Annotations: map[string]string{infrav1beta2.CreateInfrastructureAnnotation: "true"},
+						},
+					},
+				}
+			},
+			expectedError: errors.New("PowerVS zone is not set"),
+		},
+		{
+			name: "When resource group name is not set",
+			powervsClusterScope: func() *scope.PowerVSClusterScope {
+				clusterScope := &scope.PowerVSClusterScope{
+					IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+						ObjectMeta: metav1.ObjectMeta{
+							Finalizers:  []string{infrav1beta2.IBMPowerVSClusterFinalizer},
+							Annotations: map[string]string{infrav1beta2.CreateInfrastructureAnnotation: "true"},
+						},
+						Spec: infrav1beta2.IBMPowerVSClusterSpec{
+							Zone: ptr.To("dal10"),
+						},
+					},
+				}
+				mockPowerVS := powervsmock.NewMockPowerVS(gomock.NewController(t))
+				mockPowerVS.EXPECT().GetDatacenterCapabilities(gomock.Any()).Return(map[string]bool{"power-edge-router": true}, nil)
+				clusterScope.IBMPowerVSClient = mockPowerVS
+				return clusterScope
+			},
+			expectedError: errors.New("resource group name is not set"),
+		},
+		{
+			name: "When reconcile Power VS resource returns requeue",
+			powervsClusterScope: func() *scope.PowerVSClusterScope {
+				clusterScope := &scope.PowerVSClusterScope{
+					IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+						ObjectMeta: metav1.ObjectMeta{
+							Finalizers:  []string{infrav1beta2.IBMPowerVSClusterFinalizer},
+							Annotations: map[string]string{infrav1beta2.CreateInfrastructureAnnotation: "true"},
+						},
+						Spec: infrav1beta2.IBMPowerVSClusterSpec{
+							Zone: ptr.To("dal10"),
+							ResourceGroup: &infrav1beta2.IBMPowerVSResourceReference{
+								ID: ptr.To("rg-id"),
+							},
+						},
+						Status: infrav1beta2.IBMPowerVSClusterStatus{
+							ServiceInstance: &infrav1beta2.ResourceReference{
+								ID: ptr.To("serviceInstanceID"),
+							},
+						},
+					},
+				}
+				mockPowerVS := powervsmock.NewMockPowerVS(gomock.NewController(t))
+				mockPowerVS.EXPECT().GetDatacenterCapabilities(gomock.Any()).Return(map[string]bool{"power-edge-router": true}, nil)
+				clusterScope.IBMPowerVSClient = mockPowerVS
+
+				mockResourceClient := resourceclientmock.NewMockResourceController(gomock.NewController(t))
+				mockResourceClient.EXPECT().GetResourceInstance(gomock.Any()).Return(&resourcecontrollerv2.ResourceInstance{
+					Name:  ptr.To("serviceInstanceName"),
+					ID:    ptr.To("serviceInstanceID"),
+					State: ptr.To(string(infrav1beta2.ServiceInstanceStateProvisioning)),
+				}, nil, nil)
+				clusterScope.ResourceClient = mockResourceClient
+
+				mockVPC := vpcmock.NewMockVpc(gomock.NewController(t))
+				mockVPC.EXPECT().GetVPCByName(gomock.Any()).Return(nil, errors.New("vpc not found"))
+				clusterScope.IBMVPCClient = mockVPC
+
+				return clusterScope
+			},
+			expectedResult: ctrl.Result{RequeueAfter: 30 * time.Second},
+		},
+		{
+			name: "When reconcile VPC and Power VS resource returns error",
```

Comment:
maybe we can also cover the case where only requeue with true is returned without an error?

Reply:
Added a new test case where both vpc and power vs resources returns reque as true without error